vscode-dotnet-runtime-library/src/Utils/CommandExecutor.ts

```diff
@@ -0,0 +1,210 @@
+    {
+        this.eventStream.post(new CommandExecutionUnderSudoEvent(`The command ${commandFollowUps} is being ran under sudo.`));
+
+        if(this.isRunningUnderWSL())
+        {
+            // For WSL, vscode/sudo-prompt does not work.
+            // This is because it relies on pkexec or a GUI app to popup and request sudo privilege.
+            // GUI in WSL is not supported, so it will fail.
+            // We had a working implementation that opens a vscode box and gets the user password, but that will require more security analysis.
+
+            const err = new DotnetWSLSecurityError(new Error(`Automatic .NET SDK Installation is not yet supported in WSL due to VS Code & WSL limitations.
+Please install the .NET SDK manually and add it to the path by following https://github.com/dotnet/vscode-dotnet-runtime/blob/main/Documentation/troubleshooting-runtime.md#manually-installing-net`));
+            this.eventStream.post(err);
+            throw err.error;
+        }
+
+        // We wrap the exec in a promise because there is no synchronous version of the sudo exec command for vscode/sudo
+        return new Promise<string>((resolve, reject) =>
+        {
+            // The '.' character is not allowed for sudo-prompt so we use 'DotNET'
+            const options = { name: 'VS Code DotNET Acquisition' };
+            exec(commandFollowUps.join(' '), options, (error?: any, stdout?: any, stderr?: any) =>
+            {
+                let commandResultString = '';
+
+                if (stdout)
+                {
+                    commandResultString += stdout;
+                }
+                if (stderr)
+                {
+                    this.eventStream.post(new CommandExecutionStdError(`The command ${commandFollowUps} encountered stderr, continuing. ${stderr}.`));
+                    commandResultString += stderr;
+                }
+
+                if (error)
+                {
+                    this.eventStream.post(new CommandExecutionUserCompletedDialogueEvent(`The command ${commandFollowUps} failed to run under sudo.`));
+                    reject(error);
+                }
+                else
+                {
+                    this.eventStream.post(new CommandExecutionUserCompletedDialogueEvent(`The command ${commandFollowUps} successfully ran under sudo.`));
+                    resolve(commandResultString);
+                }
+            });
+        });
+    }
+
+    /**
+     *
+     * @param command The command to run as a whole string. Commands with && will be run individually. Sudo commands will request sudo from the user.
+     * @param workingDirectory - the directory to execute in. Only works for non sudo commands.
+     *
+     * @returns the result(s) of each command. Can throw generically if the command fails.
+     */
+    public async execute(command : string, options : any | null = null) : Promise<string[]>
```

Comment:
see my note on the schema document - I think you are going to want to move away from a single string for the commands here. it's a giant tarpit of failure in the medium term.